### PR TITLE
Add a timeout when initializing the Podman client (broken Podman should not affect `odo dev` on cluster)

### LINKED ISSUE #6575: Broken podman affects `odo dev` run against cluster
/kind bug

## What versions of software are you using?

**Operating System:**
MacOS  13.1.

**Output of `odo version`:**

▶ odo version
odo v3.6.0 (a03576435)

## How did you run odo exactly?

`odo dev`

## Actual behavior
command freezes with no output

verbose logging shows that it is due to `podman version`

```
▶ odo dev -v 4
I0206 13:56:31.082562    5298 implem.go:109] The path for preference file is /Users/tkral/.odo/preference.yaml
I0206 13:56:31.082659    5298 segment.go:265] Checking telemetry enable status
I0206 13:56:31.082662    5298 segment.go:295] Sending telemetry enabled by preference
I0206 13:56:31.084436    5298 version.go:28] executing [podman version --format json]
```

The experimental flag (ODO_EXPERIMENTAL_MODE) is not enabled, and no `--run-on` is specified.

## Expected behavior

Podman should not effect odo run if the cluster is used. 

## Any logs, error output, etc?

This is probably podman's fault or something on my setup. When I suspend my laptop and I have podman VM running, it sometimes gets into a  weird state where no podman commands respond (it just freezes with no output)

**When odo is executed with `--run-on=cluster` it should not try to run any `podman` commands.**

**If I would be using `--run-on=podman` then I would actually expect that it hangs as there is not much we can do from odo side about this situation.**

### PR DESCRIPTION
**Pull request by rm3l**
**What type of PR is this:**
/kind bug
/area dev

**What does this PR do / why we need it:**
This introduces a timeout for the Podman client initialization (e.g., a timeout for the `podman version` command to finish). The default value is `1s`.

For some reason, this made the Podman tests flaky on GitHub (sometimes, `podman version` would take more than 1s to return). The timeout has therefore been made configurable (via the `PODMAN_CMD_INIT_TIMEOUT` environment variable, defaulting to `1s`), and set to a slightly higher value in the tests.

**Which issue(s) this PR fixes:**
Fixes #6575

**PR acceptance criteria:**

- [ ] Unit test 

- [x] Integration test 

- [x] [Documentation](https://deploy-preview-6808--odo-docusaurus-preview.netlify.app/docs/overview/configure/#environment-variables-controlling-odo-behavior) 

**How to test changes / Special notes to the reviewer:**
This can be tested by adding a script/program to your `PATH` that would delay all calls to Podman, e.g.:
```shell
mkdir -p ~/.local/bin

cat <<EOF > ~/.local/bin/podman-with-delay
#!/bin/bash

sleep 300 && podman "$@"
EOF

export PODMAN_CMD=~/.local/bin/podman-with-delay

# If ~/.local/bin is not already in your PATH
export PATH="~/.local/bin:$PATH"
```

At this point, `odo dev` on cluster should no longer hang.

`odo dev --platform=podman` would error out, but this is expected.

### CODE REVIEW

File: pkg/podman/version.go [hunk 1]
```diff
@@ -23,20 +27,78 @@ type SystemVersionReport struct {
...
+		if rErr != nil {
+			klog.V(7).Infof("unable to read from stderr pipe: %v", rErr)
+		}
+		stderr = buf.String()
+	}()
+
+	// Wait will block until the timeout expires or the command exits. It will then close all resources associated with cmd,
+	// including the stdout and stderr pipes above, which will in turn terminate the goroutines spawned above.
+	wErr := cmd.Wait()
+	if wErr != nil {
```
[the remark this answers is not in the document]
Comment (rm3l) [reply pkg/podman/version.go:97]: From what I read, `cmd.Wait` is supposed to close the pipes, which will, in turn, unblock the goroutines. But let me wait for them as an additional safety measure..
EDIT: Done in [`2d0a448` (#6808)](https://github.com/redhat-developer/odo/pull/6808/commits/2d0a4488e48a79eff45b95b4d6755c419adb7bf2)

[the remark this answers is not in the document]
Comment (feloy) [reply pkg/podman/version.go:97]: Yes, I agree with this, but `buf.ReadFrom` and `json.NewDecoder(stdoutPipe).Decode(&result)` need some time to terminate after the pipes are closed and to have the result in `err` and `rErr`, we cannot have the guarantee this is done when Wait terminates

**Review by feloy — APPROVED** [review]

**Review by feloy — APPROVED** [review]

**Comment by rm3l** [comment]
```
Running oc.exe with args [oc create configmap config-map-for-cleanup --from-literal type=testing --from-literal team=odo -n cmd-devfile-deploy-test161cji] and odo env: []
  [oc] error: failed to create configmap: Post "[https://c114-e.eu-de.containers.cloud.ibm.com:30329/api/v1/namespaces/cmd-devfile-deploy-test161cji/configmaps?fieldManager=kubectl-create&quot;:](https://c114-e.eu-de.containers.cloud.ibm.com:30329/api/v1/namespaces/cmd-devfile-deploy-test161cji/configmaps?fieldManager=kubectl-create%22:) dial tcp 149.81.180.114:30329: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
  [FAILED] in [BeforeEach] - C:/Users/Administrator.ANSIBLE-TEST-VS/3936/tests/helper/helper_cmd_wrapper.go:101 @ 05/11/23 11:03:24.24
  Deleting project: cmd-devfile-deploy-test47osp
  Running oc.exe with args [oc delete project cmd-devfile-deploy-test47osp --wait=false] and odo env: []
  [oc] Error from server (NotFound): namespaces "cmd-devfile-deploy-test47osp" not found
  [FAILED] in [AfterEach] - C:/Users/Administrator.ANSIBLE-TEST-VS/3936/tests/helper/helper_cmd_wrapper.go:101 @ 05/11/23 11:03:24.523
  << Timeline
```

Network errors.

/override windows-integration-test/Windows-test

**Comment by rm3l** [comment]
```
dial tcp: lookup c100-e.eu-de.containers.cloud.ibm.com

...
Summarizing 1 Failure:
  [FAIL] odo remove binding command tests when the component with binding is bootstrapped (bindingName=my-nodejs-app-cluster-sample-ocp) when odo dev is running when binding is removed [It] should have led odo dev to delete ServiceBinding from the cluster
  /go/odo_1/tests/integration/cmd_remove_binding_test.go:73

Ran 486 of 886 Specs in 1788.480 seconds
FAIL! -- 485 Passed | 1 Failed | 0 Pending | 400 Skipped
```

Previous run passed.

/override OpenShift-Integration-tests/OpenShift-Integration-tests

**Comment by rm3l** [comment]
/override kubernetes-infra-stage-test

Not related.

**Pull request merged**